src/excmds.ts

```diff
@@ -3089,48 +3088,63 @@ export function yank(...content: string[]) {
 }
 
 /**
- * Copies a string to the clipboard/selection buffer depending on the user's preferences
+ * Copies a string to the clipboard/selection buffer depending on the user's preferences.
  *
  * @hidden
  */
 //#background_helper
-async function setclip(str) {
-    // Functions to avoid retyping everything everywhere
+async function setclip(data: string) {
+    // Function to avoid retyping everything everywhere
+    const setclip_selection = () => Native.clipboard("set", data)
 
-    // Note: We're using fillcmdline here because exceptions are somehow not caught. We're rethrowing because otherwise the error message will be overwritten with the "yank successful" message.
-    const s = () => Native.clipboard("set", str)
-    const c = () => messageActiveTab("commandline_frame", "setClipboard", [str])
-
-    let promises = []
+    let promises: Array<Promise<any>> = []
     switch (await config.getAsync("yankto")) {
         case "selection":
-            promises = [s()]
+            promises = [setclip_selection()]
             break
         case "clipboard":
-            promises = [c()]
+            promises = [setclip_api(data)]
             break
         case "both":
-            promises = [s(), c()]
+            promises = [setclip_selection(), setclip_api(data)]
             break
     }
     return Promise.all(promises)
 }
 
+/**
+ * Copies a string to the clipboard using the Clipboard API.
+ * @hidden
+ */
+//#background_helper
```

Comment:
It was surprising to me that this was a background command. It's presumably because of this:

> When using content scripts, the Clipboard API is available only for HTTPS pages. As a workaround, use messaging between your content scripts and the background script.

which we should document, I think : )

Reply:
Well, about that. I first tried `content_helper` – I needed to access `window`, so that seemed the right place to go –, but that gave me a compile error. So I tried `content`, which gave me a run time `TypeError: <something> is not a function`. It just happened to work with `background_helper` when I tried it.

Reply:
By which I mean: I don't really understand how these macros work, I just winged it.

Reply:
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Interact_with_the_clipboard is the page that tells you why you got a runtime error : )

No idea why you got a compile error with content_helper.

Reply:
Hmm. I thought I'd tried it on Tridactyl's new tab page, but I guess it must've been on a non-https page, unless `moz-extension://` doesn't count as https.
The error I got also didn't look like something Firefox would throw in that case, though. I'd assume Firefox would either throw something like a generic `Error: the Clipboard API is only available on HTTPS connections`, a `TypeError: window.navigator.clipboard.writeText is not a function`, or a `TypeError: window.navigator.clipboard is undefined`, but I remember that wasn't it. I think it was something along the lines of `TypeError: object[foobar] is not a function`.

Reply:
oh, it's because `setclip` is a background function and you can't call content commands from the background.